File: src/main/solved/Programmers/level2/java/구명보트.java
package Programmers.level2.java;

import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;

        if(people.length == 1) return 1;

        Arrays.sort(people);
        int start = 0, end = people.length-1;
        while(start <= end){
            if(people[start]+people[end] > limit){
                end -= 1; answer+=1;
            }else{
                start+=1; end-=1;
                answer+=1;
            }
        }
        return answer;
    }
}

File: src/main/solved/Programmers/level2/python/구명보트.py
def solution(people, limit):
    answer = 0

    people.sort()
    left, right = 0, len(people)-1

    while left < right:
        weightSum = people[left]

        while left < right and weightSum + people[right] > limit:
            answer += 1
            right -= 1

        left += 1
        answer += 1
        right -= 1

    if left == right:
        answer += 1

    return answer